sequencer/src/persistence/fs.rs

```diff
@@ -1277,6 +1302,37 @@ impl SequencerPersistence for Persistence {
 
         Ok(result)
     }
+
+    async fn load_state_cert(&self) -> anyhow::Result<LightClientStateUpdateCertificate<SeqTypes>> {
+        let inner = self.inner.read().await;
+        let dir_path = inner.light_client_state_update_certificate_dir_path();
+
+        let mut result = LightClientStateUpdateCertificate::genesis();
+
+        if !dir_path.is_dir() {
+            return Ok(result);
+        }
+        for (epoch, path) in epoch_files(dir_path)? {
+            if epoch <= result.epoch {
+                continue;
+            }
+            let bytes = fs::read(&path).context(format!(
+                "reading light client state update certificate {}",
+                path.display()
+            ))?;
+            let cert = bincode::deserialize::<LightClientStateUpdateCertificate<SeqTypes>>(&bytes)
+                .context(format!(
+                    "parsing light client state update certificate {}",
+                    path.display()
+                ))?;
+
+            if cert.epoch > result.epoch {
```

Comment:
why deserialize the cert and then have this check? There is already an epoch from the sub directory. Maybe this check should be moved up and have another check that sub dir epoch == cert.epoch

Reply:
Yes this is redundant, [60ddcad](https://github.com/EspressoSystems/espresso-network/pull/2762/commits/60ddcada8e0f37d832f2a9322726839109c12964)